app/(farmer)/journal/page.tsx

```diff
@@ -32,29 +40,58 @@ export default function FarmerJournal() {
     resolver: zodResolver(step.schema)
   });
 
-  const onSubmit = methods.handleSubmit((data) => {
-    if (isLastStep) console.log(data);
-    else next();
-  }, console.log);
+  const onBack = () => {
+    setError(false)
+    back()
+    ref.current?.scrollTo({ top: 0 })
+  }
+
+  const onSubmit = methods.handleSubmit(async (data) => {
+    setError(false)
+    if (!isLastStep) {
+      next()
+      ref.current?.scrollTo({ top: 0 })
+      return
+    }
+    const processed = processFormData(data)
+    const response = await fetch("/api/drupal/node/farmer_daily_journal", {
+      method: "POST",
+      body: JSON.stringify(processed)
+    })
```

Comment:
Should this use `jsonApiFetch`? Just so that all `fetches` run through the same function?

Reply:
The `jsonApiFetch` runs on the server and automatically attaches the auth token to the GET requests.
However, it isn't the case for the  POST requests, you must use route handlers.

Alternatively, I could use server action, but it's still experimental and unstable.